File: Data Structures & Algorithms/buy-and-sell-crypto/submission-5.js
class Solution {
    /**
     * @param {number[]} prices
     * @return {number}
     */
    maxProfit(prices) {
        let maxProfit = 0
        let l = 0
        let r = 1

        while (r < prices.length){
            if (prices[l] > prices[r]){
                l = r
                r += 1
            }else if (prices[l] < prices[r]){
                maxProfit = Math.max(maxProfit, prices[r] - prices[l])
                r += 1
            } else {
                l = r
                r += 1
            }
            
        }

        return maxProfit
    }
}

File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-5.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        maxLength = 1
        l, r = 0, 0

        freqMap = {}
        while r < len(s):
            freqMap[s[r]] = freqMap.get(s[r], 0) + 1
            while r - l + 1 - max(freqMap.values()) > k:
                freqMap[s[l]] -= 1
                l += 1
            maxLength = max(maxLength, r - l + 1)
            r += 1
           
            








        return maxLength
        